annotation/copy_test.go

```diff
@@ -0,0 +1,121 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotation
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type CopyTestSuite struct {
+	suite.Suite
+	initStructs   []any
```

Comment:
We can probably make this a generic struct with a type constraint on having `Copy` method to remove some boilerplate code below.

Reply:
Created issue #172 to perform this refactoring in a follow-up PR.